### PR TITLE
ion-item button with ripple

### PR DESCRIPTION
**Pull request by antoinecfmws**
Currently the documentation is not showing how to create an ion-item button and show the ripple. This commit simply adds an example to add an ion-item with the attribute button to show a ripple.

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [N/A] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [N/A] Build (`npm run build`) was run locally and any changes were pushed
- [N/A ] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [X] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?
N/A

Issue Number: N/A

## What is the new behavior?
Simply modifies the documentation to add the "button" attribute to an ion-item.

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

N/A

## Other information

N/A

### CODE REVIEW

**Comment by liamdebeasi** [comment]
Thanks for the PR! In order to update the docs, there are a few things you will need to do:

1. You will need to provide usage examples in the `angular.md`, `javascript.md`, `react.md`, and `vue.md` files. I can help with usage if you are unsure about the syntax in other frameworks. The `readme.md` file is autogenerated, so your examples should be placed in these files, otherwise they will be overwritten.
2. Run `npm run lint.fix`.
3. Run `npm run build`. This will generate a new `readme.md` file.

After that, you can commit and push. Please let me know if you have any questions.

**Comment by antoinecfmws** [comment]
@liamdebeasi thank you, just updated it.

**Comment by antoinecfmws** [comment]
We do have currently click events across angular, vue, and react. Do you want to remove them all for the "Item" component?

File: core/src/components/item/usage/javascript.md [hunk 1]
```diff
@@ -15,6 +15,13 @@ Basic Usage
   </ion-label>
 </ion-item>
 
+<!-- Item as a Button with Ripple -->
+<ion-item (click)="buttonClick()" button>
```
Comment (liamdebeasi) [inline core/src/components/item/usage/javascript.md]:
```suggestion
<ion-item onclick="buttonClick()" button>
```
`(click)` is Angular-only

File: core/src/components/item/usage/vue.md [hunk 2]
```diff
@@ -16,6 +16,13 @@ Basic Usage
     </ion-label>
   </ion-item>
 
+  <!-- Item as a Button with Ripple -->
+  <ion-item (click)="buttonClick()" button>
```
Comment (liamdebeasi) [inline core/src/components/item/usage/vue.md]:
```suggestion
  <ion-item @click="buttonClick()" button>
```

**Comment by liamdebeasi** [comment]
Oh I see what you're saying. In that case let's just keep it for now. I made a few more comments.

**Comment by liamdebeasi** [comment]
You'll also need to run `npm run lint.fix && npm run build` again and push any changes. (This will update the `readme.md` file

**Comment by antoinecfmws** [comment]
I regenerated readme.md for component item. 

For some reason, both `npm run lint.fix` and `npm run build` on windows, generates a list of 100+ files that are supposedly changed, but when comparing them they are the same. I sent a question on slack but please let me know if you have any suggestions. It's hard sometimes to find which of these files have changed, unless you are fine with me committing all next time..

**Comment by brandyscarney** [comment]
Thanks so much for the PR! Instead of adding a new item, I updated the existing items so that if they have a click event they will get the `button` property. This was an oversight by us, but any item with a click event should have it. 🙂 https://ionicframework.com/docs/api/item#clickable-items

**Comment by antoinecfmws** [comment]
Makes sense thanks!

**Comment by liamdebeasi** [comment]
Merged. Thank you!

**Pull request merged**